### PR TITLE
feat: add missing fields in ApplicationInfo (#1190)

### LINKED ISSUE #1190: Add missing fields in the `ApplicationInfo` object
**Feature Description:** Many fields are missing
**Justification:** See https://discord.com/developers/docs/resources/application#application-object

### PR DESCRIPTION
**Pull request by Azn9**
Closes #1190 
Requires https://github.com/Discord4J/discord-json/pull/179

Depends on https://github.com/Discord4J/Discord4J/pull/1183

### CODE REVIEW

File: core/src/main/java/discord4j/core/object/ApplicationInstallParams.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public List<String> getScopes() {
+        return this.data.scopes();
+    }
+
+    /**
+     * Gets the permissions of this configuration
+     *
+     * @return The permissions of this configuration
+     */
+    public String getPermissions() {
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/object/ApplicationInstallParams.java:43]: This need to be parsed in the enum permission?

File: core/src/main/java/discord4j/core/object/ApplicationIntegrationTypeConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            .map(InstallParamsData::scopes)
+            .orElse(Collections.emptyList());
+    }
+
+    /**
+     * Gets the permissions of this configuration
+     *
+     * @return The permissions of this configuration
+     */
+    public Optional<String> permissions() {
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/object/ApplicationIntegrationTypeConfiguration.java:49]: This need to be parsed in the enum permission?

File: core/src/main/java/discord4j/core/object/ApplicationTeamMember.java [hunk 3]
```diff
@@ -117,7 +117,7 @@ public Mono<User> getUser(EntityRetrievalStrategy retrievalStrategy) {
      * @return The user's role on the team.
      */
     public TeamMemberRole getRole() {
-        return TeamMemberRole.of(data.role());
+        return data.role().toOptional().map(TeamMemberRole::of).orElse(TeamMemberRole.ADMIN);
```
Comment (Doc94) [inline core/src/main/java/discord4j/core/object/ApplicationTeamMember.java:120]: in what cases this info not come?

**Review by Doc94 — CHANGES_REQUESTED** [review]

File: core/src/main/java/discord4j/core/object/ApplicationTeamMember.java (hunk 3 above)
Comment (Azn9) [reply core/src/main/java/discord4j/core/object/ApplicationTeamMember.java:120]: If the user is the owner of the team, he don't have any role

Comment (Doc94) [reply core/src/main/java/discord4j/core/object/ApplicationTeamMember.java:120]: ohhh that case...  yeah unless add a OWNER role but ADMIN works too then.

**Pull request merged**